src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -67,16 +69,29 @@ private function registerHandlers(ContainerBuilder $container)
 
         foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
             foreach ($tags as $tag) {
-                $handles = $tag['handles'] ?? $this->guessHandledClass($r = $container->getReflectionClass($container->getParameterBag()->resolveValue($container->getDefinition($serviceId)->getClass())), $serviceId);
-
-                if (!class_exists($handles)) {
-                    $messageClassLocation = isset($tag['handles']) ? 'declared in your tag attribute "handles"' : sprintf('used as argument type in method "%s::__invoke()"', $r->getName());
-
-                    throw new RuntimeException(sprintf('Invalid handler service "%s": message class "%s" %s does not exist.', $serviceId, $handles, $messageClassLocation));
+                $handles = $tag['handles'] ?? $this->guessHandledClasses($r = $container->getReflectionClass($container->getDefinition($serviceId)->getClass()), $serviceId);
+                if (!$handles) {
+                    throw new RuntimeException(sprintf('The handler service "%s" does not seem to handle any message.', $serviceId));
```

Comment:
This exception should be moved to `guessHandledClasses` and message updated for better DX, as it means the `MessageSubscriberInterface` was wrongly implemented by returning an empty array.

Reply:
@ogizanagi updated accordingly 👍